src/main/java/org/opensearch/security/tools/democonfig/SecuritySettingsConfigurer.java

```diff
@@ -0,0 +1,316 @@
+            if (shouldValidatePassword
+                && !ADMIN_PASSWORD.isEmpty()
+                && passwordValidator.validate("admin", ADMIN_PASSWORD) != RequestContentValidator.ValidationError.NONE) {
+                System.out.println("Password " + ADMIN_PASSWORD + " is weak. Please re-try with a stronger password.");
+                System.exit(-1);
+            }
+
+            // if ADMIN_PASSWORD is still an empty string, it implies no custom password was provided. We proceed with generating a new one.
+            if (ADMIN_PASSWORD.isEmpty()) {
+                System.out.println("No custom admin password found. Generating a new password now.");
+                // generate a new random password
+                // We always validate a generated password
+                while (passwordValidator.validate("admin", ADMIN_PASSWORD) != RequestContentValidator.ValidationError.NONE) {
+                    ADMIN_PASSWORD = generatePassword();
+                }
+            }
+
+            // print the password to the logs
+            System.out.println("\t***************************************************");
+            System.out.println("\t\tADMIN PASSWORD SET TO: " + ADMIN_PASSWORD);
+            System.out.println("\t***************************************************");
+
+            writePasswordToInternalUsersFile(ADMIN_PASSWORD, INTERNAL_USERS_FILE_PATH);
+
+        } catch (IOException e) {
+            System.out.println("Exception: " + e.getMessage());
+            System.exit(-1);
+        }
+    }
+
+    /**
+     * Generate password hash and update it in the internal_users.yml file
+     * @param adminPassword the password to be hashed and updated
+     * @param internalUsersFile the file path string to internal_users.yml file
+     * @throws IOException while reading, writing to files
+     */
+    static void writePasswordToInternalUsersFile(String adminPassword, String internalUsersFile) throws IOException {
+        String hashedAdminPassword = Hasher.hash(adminPassword.toCharArray());
+
+        if (hashedAdminPassword.isEmpty()) {
+            System.out.println("Hash the admin password failure, see console for details");
+            System.exit(-1);
+        }
+
+        Path tempFilePath = Paths.get(internalUsersFile + ".tmp");
+        Path internalUsersPath = Paths.get(internalUsersFile);
+
+        try (
+            BufferedReader reader = new BufferedReader(new FileReader(internalUsersFile, StandardCharsets.UTF_8));
+            BufferedWriter writer = new BufferedWriter(new FileWriter(tempFilePath.toFile(), StandardCharsets.UTF_8))
+        ) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                if (line.matches(" *hash: *\"\\$2a\\$12\\$VcCDgh2NDk07JGN0rjGbM.Ad41qVR/YFJcgHp0UGns5JDymv..TOG\"")) {
+                    line = line.replace(
```

Comment:
I assume that this is the hash original admin password, which is "admin" at the moment. I know that we discussed this before, and I still think that if we are deciding to implement this entire new user flow for admin password setup, we should remove the current default admin credential from the `internal_users.yml`. Wdyt? (Btw even if we decide to remove it, I think this can be resolved as a followup).

Reply:
this can be done in future iterations. The purpose of this iteration was to translate it to a java tool. I can definitely add it in a follow-up PR.

Reply:
I think we have unit tests which require the old default password be valid. If there is someway to replace this or this is not the case, I agree with Ryan.

Reply:
I don't think we use script to initiate configuration in unit tests.  This will only be used when spinning up a distribution and installing demo configuration.

Also, for tests I have added -t option which skips password validation.

Reply:
Yes I agree, and lets do this in the future iterations.